frontend/test/pytest/test_transform.py

```diff
@@ -121,44 +166,896 @@ def qfunc(data, x, weights):
     assert expected_shape == observed_shape
 
 
-def test_split_non_commuting(backend):
-    """Test split non commuting"""
+@pytest.mark.xfail(reason="JIT and QJIT give different results.")
+def test_batch_partial(backend):
+    """Test batch partial"""
 
     def qnode_builder(device_name):
         """Builder"""
 
-        @qml.transforms.split_non_commuting
-        @qml.qnode(qml.device(device_name, wires=6), interface="jax")
+        @qml.qnode(qml.device(device_name, wires=2), interface="jax")
+        def qfunc(x, y):
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            return qml.expval(qml.Z(0) @ qml.Z(1))
+
+        return qfunc
+
+    batch_size = 4
+    x = np.linspace(0.1, 0.5, batch_size)
+    y = np.array(0.2)
+
+    qnode_control = qml.batch_partial(qnode_builder("default.qubit"), y=y)
+    qnode_backend = qml.batch_partial(qnode_builder(backend), y=y)
+
+    jax_jit = jax.jit(qnode_control)
+    compiled = qjit(qnode_backend)
+    expected = jax_jit(x)
+    observed = compiled(x)
+    assert np.allclose(expected, observed)
+
+    _, expected_shape = jax.tree_util.tree_flatten(expected)
+    _, observed_shape = jax.tree_util.tree_flatten(observed)
+    assert expected_shape == observed_shape
+
+
+def test_cancel_inverses(backend):
+    """Test cancel_inverses"""
+
+    def qnode_builder(device_name):
+        """Builder"""
+
+        @qml.transforms.cancel_inverses
+        @qml.qnode(qml.device(device_name, wires=3), interface="jax")
+        def qfunc(x, y, z):
+            qml.Hadamard(wires=0)
+            qml.Hadamard(wires=1)
+            qml.Hadamard(wires=0)
+            qml.RX(x, wires=2)
+            qml.RY(y, wires=1)
+            qml.X(1)
+            qml.RZ(z, wires=0)
+            qml.RX(y, wires=2)
+            qml.CNOT(wires=[0, 2])
+            qml.X(1)
+            return qml.expval(qml.Z(0))
+
+        return qfunc
+
+    qnode_control = qnode_builder("default.qubit")
+    qnode_backend = qnode_builder(backend)
+
+    jax_jit = jax.jit(qnode_control)
+    compiled = qjit(qnode_backend)
+
+    x, y, z = 0.1, 0.2, 0.3
+    expected = jax_jit(x, y, z)
+    observed = compiled(x, y, z)
+    assert np.allclose(expected, observed)
+
+    _, expected_shape = jax.tree_util.tree_flatten(expected)
+    _, observed_shape = jax.tree_util.tree_flatten(observed)
+    assert expected_shape == observed_shape
+
+
+@pytest.mark.xfail(reason="Not JAX JIT compatible")
```

Comment:
In cases where jax jit fails but qjit succeeds, I think we should compare qjit with plain pennylane.

Reply:
In this case jax jit and qjit both fail for the same reason. I think just giving a more informative xfail reason is sufficient.